library/core/src/primitive_docs.rs

```diff
@@ -1506,6 +1506,110 @@ mod prim_ref {}
 /// Note that all of this is not portable to platforms where function pointers and data pointers
 /// have different sizes.
 ///
+/// ### ABI compatibility
+///
+/// Generally, when a function is declared with one signature and called via a function pointer with
```

Comment:
Where do we document and guarantee what one has to do when linking multiple Rust objects together? Do we even support that with `extern "Rust"` functions?

We have to put these docs somewhere, and function pointers are the only way to trigger these issues inside the language (without exotic things such as `dlopen` or manually linking things together), so I figured this would make sense. If you can think of a better place where we can put this, I can move it.

Reply:
I understand from seeing other issues why you see "inside the language" as a useful initial scope because that's what seems to matter for the SIMD stuff. I don't object to that.

IME people are much more likely to run into ABI issues in cross-language situations since there are no guardrails at all, so I hope we at least are open to solving the issue for cross-language cases too. Perhaps that means adding similar language to the documentation of `extern` and then factoring out the commonality to some top-level section of the language reference, in a future PR.

Reply:
> IME people are much more likely to run into ABI issues in cross-language situations

Fully agreed. It's also a much less defined space, you basically have to define "which C type is compatible with which Rust type" (and then potentially also which C++/Julia/whatever type, though I guess we can rely on those languages saying which C types *their* types are compatible with). This depends on the target and honestly I'm quickly out of my depth for those questions. ABI still surprises me with new nightmares every few weeks, so I'm sure there's more to come.

I do hope someone will pick this up eventually.

> I hope we at least are open to solving the issue for cross-language cases too.

Of course we are, I didn't want to give any indication to the contrary! It's just not the problem I want to solve right now. It's not on my own personal todo list either, at the moment.